Modify rule S2259: Expand and adjust for LaYC (#2939)
SonarSource/rspec: rules/S3807/cfamily/rule.adoc

* CWE - https://cwe.mitre.org/data/definitions/476[476 NULL Pointer Dereference]
* CERT - https://wiki.sei.cmu.edu/confluence/x/aDdGBQ[EXP01-J. Do not use a null in a case where an object is required]

=== Related rules

> * S2259 detects dereferences of null pointer

Reviewer: ```suggestion
* S2259 detects dereferences of null pointer
```
```suggestion
* S2259 detects dereferences of null pointers
```

> * S2637 detects passing of null pointer as argument for parameter that is annotated with `nonnull`

Reviewer: ```suggestion
* S2637 detects uses of null pointers as arguments for parameters that are annotated with `nonnull`
```
